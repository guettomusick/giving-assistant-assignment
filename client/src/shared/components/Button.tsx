import styled from 'styled-components';

const Button = styled.button`
  font-size: 1rem;
  color: white;
  background-color: #18A0FB;
  padding: 10px 20px;
  border: none;
  border-radius: 6px;
  outline : none;

  ${props => !props.disabled && `
    cursor: pointer;

    &:hover {
      background-color: #0890EB;
    }
  `};
`;

export default Button;