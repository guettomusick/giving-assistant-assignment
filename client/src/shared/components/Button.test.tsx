import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

test('renders', () => {
  const component = renderer.create(
    <Button onClick={ () => {} }>Button</Button>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
