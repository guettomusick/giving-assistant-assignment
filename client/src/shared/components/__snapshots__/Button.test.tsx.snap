// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  className="sc-AxjAm dLmywG"
  onClick={[Function]}
>
  Button
</button>
`;
